rubix-bookkeeper/src/main/java/com/qubole/rubix/bookkeeper/utils/DiskUtils.java

```diff
@@ -79,10 +80,37 @@ public static int getCacheSizeMB(Configuration conf)
 
     long cacheSize = 0;
     for (int disk = 0; disk < diskMap.size(); disk++) {
-      long cacheDirSize = FileUtils.sizeOfDirectory(new File(diskMap.get(disk) + cacheDirSuffix));
+      long cacheDirSize = getDirSizeMB(new File(diskMap.get(disk) + cacheDirSuffix));
       cacheSize += cacheDirSize;
     }
-    return (int) DiskUtils.bytesToMB(cacheSize);
+    return (int) cacheSize;
+  }
+
+  /**
+   * Gets the actual size occupied on the disk, for the given directory using du command.
+   *
+   * @return The size of the cache in MB.
+   */
+  public static long getDirSizeMB(File dirname)
+  {
+    String cmd = "du -s " + dirname.toString();
+    StringBuffer output = new StringBuffer();
+
+    try {
+      Process p;
+      String[] env = new String[] {"BLOCKSIZE=1024"};
+      p = Runtime.getRuntime().exec(cmd, env);
+      p.waitFor();
+      BufferedReader reader = new BufferedReader(new InputStreamReader(p.getInputStream()));
+      String line = "";
+      while ((line = reader.readLine()) != null) {
+        output.append(line + "\n");
+      }
+    }
+    catch (Exception e) {
+      log.error("Exception while calculating the size of the folder " + dirname.toString());
+    }
+    return Long.parseLong(output.toString().split("\\s+")[0]) / 1024;
```

Comment:
Exception from the build logs:
```java
java.lang.NumberFormatException: For input string: ""
	at java.lang.NumberFormatException.forInputString(NumberFormatException.java:65)
	at java.lang.Long.parseLong(Long.java:453)
	at java.lang.Long.parseLong(Long.java:483)
	at com.qubole.rubix.bookkeeper.utils.DiskUtils.getDirSizeMB(DiskUtils.java:113)
	at com.qubole.rubix.bookkeeper.utils.DiskUtils.getCacheSizeMB(DiskUtils.java:83)
	at com.qubole.rubix.bookkeeper.BookKeeper$3.getValue(BookKeeper.java:197)
	at com.qubole.rubix.bookkeeper.BookKeeper$3.getValue(BookKeeper.java:193)
	at com.readytalk.metrics.StatsDReporter.reportGauge(StatsDReporter.java:250)
	at com.readytalk.metrics.StatsDReporter.report(StatsDReporter.java:175)
	at com.codahale.metrics.ScheduledReporter.report(ScheduledReporter.java:162)
	at com.codahale.metrics.ScheduledReporter$1.run(ScheduledReporter.java:117)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:473)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:304)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:178)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:293)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1152)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:622)
	at java.lang.Thread.run(Thread.java:748)
```

Use Apache Commons `NumberUtils.toInt()` to guard against empty and null strings.